dali/test/python/test_dali_cpu_only.py

```diff
@@ -0,0 +1,872 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.tfrecord as tfrec
+from test_utils import get_dali_extra_path, check_batch, RandomlyShapedDataIterator, dali_type
+from PIL import Image, ImageEnhance
+
+import numpy as np
+from nose.tools import assert_raises
+import os
+import glob
+from math import ceil, sqrt
+import tempfile
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+audio_dir = os.path.join(data_root, 'db', 'audio')
+caffe_dir = os.path.join(data_root, 'db', 'lmdb')
+caffe2_dir = os.path.join(data_root, 'db', 'c2lmdb')
+recordio_dir = os.path.join(data_root, 'db', 'recordio')
+tfrecord_dir = os.path.join(data_root, 'db', 'tfrecord')
+coco_dir = os.path.join(data_root, 'db', 'coco', 'images')
+coco_annotation = os.path.join(data_root, 'db', 'coco', 'instances.json')
+sequence_dir = os.path.join(data_root, 'db', 'sequence', 'frames')
+
+batch_size = 2
+test_data_shape = [10, 20, 3]
+def get_data():
+    out = [np.random.randint(0, 255, size = test_data_shape, dtype = np.uint8) for _ in range(batch_size)]
+    return out
+
+def test_gpu_op_bad_device():
+    test_data_shape = [1, 3, 0, 4]
+    def get_data():
+        out = [np.empty(test_data_shape, dtype = np.uint8) for _ in range(batch_size)]
+        return out
+
+    pipe = Pipeline(batch_size=batch_size, num_threads=3, device_id=-1)
+    outs = fn.external_source(source = get_data, device = "gpu")
+    pipe.set_outputs(outs)
+    assert_raises(RuntimeError, pipe.build)
+
+def test_gpu_op_bad_device_2():
```

Comment:
Seems like this is an exact copy of the previous test, except the device id. Maybe you can use a loop and yield?
Also, why is device_id = 0 raising an error? Are we running this in a machine without a GPU?

Reply:
Yes, this is the intention of the test. To run on CPU only node.

Reply:
Done